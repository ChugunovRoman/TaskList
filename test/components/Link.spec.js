'use strict';

import React from 'react';
import Link from '../../src/components/Link';

const setup = (setupProps = {}) => {
    const defaultProps = {
        active: false,
        children: 'active',
        onClick: jest.fn()
    };

    const props = { ...defaultProps, ...setupProps };
    const wrapper = mount (
        <Link {...props} />
    );

    return {
        props,
        wrapper
    };
};

describe('Components: <Link />', () => {

    it('renders without crashing', () => {
        const { wrapper } = setup();

        expect(wrapper.find('.link').hasClass('link_active')).toEqual(false);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders active link', () => {
        const { wrapper } = setup({ active: true });

        expect(wrapper.find('.link').hasClass('link_active')).toEqual(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('calls onClick() on click', () => {
        const preventDefault = jest.fn();
        const { props, wrapper } = setup();

        expect(wrapper).toMatchSnapshot();

        wrapper.find('a').simulate('click', { preventDefault });
        expect(props.onClick).toBeCalled();
        expect(preventDefault).toBeCalled();
    });

});
