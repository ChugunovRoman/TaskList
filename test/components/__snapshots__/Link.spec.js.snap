// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: <Link /> calls onClick() on click 1`] = `
<Link
  active={false}
  onClick={[Function]}
>
  <a
    className="link"
    href="#"
    onClick={[Function]}
  >
    active
  </a>
</Link>
`;

exports[`Components: <Link /> calls onClick() on click 2`] = `
<Link
  active={true}
  onClick={[Function]}
>
  <a
    className="link link_active"
    href="#"
    onClick={[Function]}
  >
    active
  </a>
</Link>
`;

exports[`Components: <Link /> renders active link 1`] = `
<Link
  active={true}
  onClick={[Function]}
>
  <a
    className="link link_active"
    href="#"
    onClick={[Function]}
  >
    active
  </a>
</Link>
`;

exports[`Components: <Link /> renders without crashing 1`] = `
<Link
  active={false}
  onClick={[Function]}
>
  <a
    className="link"
    href="#"
    onClick={[Function]}
  >
    active
  </a>
</Link>
`;
